net/dtlslistener.go

```diff
@@ -39,14 +92,56 @@ func NewDTLSListener(network string, addr string, dtlsCfg *dtls.Config) (*DTLSLi
 		return ctx, cancel
 	}
 
-	listener, err := dtls.Listen(network, a, dtlsCfg)
+	lc := udp.ListenConfig{
+		AcceptFilter: tlsPacketFilter,
+	}
+	l.listener, err = lc.Listen(network, a)
 	if err != nil {
-		return nil, fmt.Errorf("cannot create new dtls listener: %w", err)
+		return nil, err
 	}
-	l.listener = listener
+	l.wg.Add(1)
+	go l.run()
 	return &l, nil
 }
 
+func (l *DTLSListener) send(conn net.Conn, err error) {
+	select {
+	case <-l.done:
+	case l.acceptedConnChan <- acceptedConn{
+		conn: conn,
+		err:  err,
+	}:
+	}
+}
+
+func (l *DTLSListener) accept() error {
+	c, err := l.listener.Accept()
+	if err != nil {
+		l.send(nil, err)
+		return err
+	}
+	err = l.goPool(func() {
+		l.send(dtls.Server(c, l.config))
+	})
+	if err != nil {
+		_ = c.Close()
+	}
+	return err
+}
+
+func (l *DTLSListener) run() {
+	defer l.wg.Done()
+	for {
+		if l.closed.Load() {
+			return
+		}
+		err := l.accept()
+		if errors.Is(err, udp.ErrClosedListener) {
```

Comment:
Maybe we should at least log other errors? Everything except `ErrClosedListener` will be ignored, which seems to me like an information loss.

Reply:
The `pion/udp` listener send's multiple errors like `io/timeout` `context canceled` context deadline exceeded` so they are not interesting -> we want to call accept until listen connection is closed - similar as is in TCP.